tests/test_base.py

```diff
@@ -686,6 +701,292 @@ def test_namedtuple_conformation(self, namedtuple_spec: Spec):
+            (("num", 5), 5),
+            (("num", 3.14), 3.14),
+            (("num", -10), -10),
+        ],
+    )
+    def test_tagged_conformation(self, tag_spec: Spec, expected, v):
+        assert expected == tag_spec.conform(v)
+
+    @pytest.mark.parametrize(
+        "v", [None, {}, set(), [], "500x", "Just a sentence", b"500", b"byteword"]
+    )
+    def test_tagged_conformation_failure(self, tag_spec: Spec, v):
+        assert INVALID is tag_spec.conform(v)
+        assert INVALID is tag_spec.conform_valid(v)
+
+
+class TestAnySpecWithOuterConformation:
+    @pytest.fixture
+    def spec(self) -> Spec:
+        return s.any(
+            s.num("num"),
+            s.str("numstr", regex=r"\d+", conformer=int),
+            conformer=lambda v: v + 5,
+        )
+
+    @pytest.mark.parametrize(
+        "expected,v", [(10, "5"), (10, 5), (8.14, 3.14), (-5, -10)]
+    )
+    def test_conformation(self, spec: Spec, expected, v):
+        assert expected == spec.conform(v)
+
+    @pytest.mark.parametrize(
+        "v", [None, {}, set(), [], "500x", "Just a sentence", b"500", b"byteword"]
+    )
+    def test_conformation_failure(self, spec: Spec, v):
+        assert INVALID is spec.conform(v)
+        assert INVALID is spec.conform_valid(v)
+
+    @pytest.fixture
+    def tag_spec(self) -> Spec:
+        return s.any(
+            s.num("num"),
+            s.str("numstr", regex=r"\d+", conformer=int),
+            tag_conformed=True,
+            conformer=lambda v: v + 5,
+        )
+
+    @pytest.mark.parametrize(
+        "expected,v",
+        [
+            (("numstr", 10), "5"),
+            (("num", 10), 5),
+            (("num", 8.14), 3.14),
+            (("num", -5), -10),
+        ],
+    )
+    def test_tagged_conformation(self, tag_spec: Spec, expected, v):
+        assert expected == tag_spec.conform(v)
+
+    @pytest.mark.parametrize(
+        "v", [None, {}, set(), [], "500x", "Just a sentence", b"500", b"byteword"]
+    )
+    def test_tagged_conformation_failure(self, tag_spec: Spec, v):
+        assert INVALID is tag_spec.conform(v)
+        assert INVALID is tag_spec.conform_valid(v)
+
+
+class TestMergeSpecConstruction:
+    def test_merge_spec_must_have_pred(self):
+        with pytest.raises(TypeError):
+            s.merge()
+
+        with pytest.raises(ValueError):
+            s.merge("with_tag")
+
+    def test_passthrough_spec(self):
+        assert s.merge(str).is_valid("something")
+        assert s.merge("with_tag", str).is_valid("something")
+
+    def test_must_all_be_mapping_specs(self):
+        with pytest.raises(TypeError):
+            s.merge({"id": int}, s.str("name"))
+
+        with pytest.raises(TypeError):
+            s.merge("with_tag", s.num(min_=0), {"name": str})
+
+    def test_no_overlapping_required_and_optional(self):
+        with pytest.raises(KeyError):
+            s.merge(
+                {"id": int, "name": str}, {"id": lambda v: v > 0, s.opt("name"): str}
+            )
```

Comment:
Is it worth adding a test where only the required or optional keys (but not both) overlap?